src/LtiOidcLogin.php

```diff
@@ -14,36 +15,19 @@ class LtiOidcLogin
     public const ERROR_MSG_ISSUER = 'Could not find issuer';
     public const ERROR_MSG_LOGIN_HINT = 'Could not find login hint';
 
-    /**
-     * @todo Type these in v6
-     */
-    private $db;
-    private $cache;
-    private $cookie;
-
-    /**
-     * Constructor.
-     *
-     * @param  IDatabase  $database Instance of the Database interface used for looking up registrations and deployments
-     * @param  ICache  $cache    instance of the Cache interface used to loading and storing launches
-     * @param  ICookie  $cookie   instance of the Cookie interface used to set and read cookies
-     */
-    public function __construct(IDatabase $database, ?ICache $cache = null, ?ICookie $cookie = null)
+    public function __construct(
+        public IDatabase $db,
+        public ICache $cache,
+        public ICookie $cookie)
     {
-        /**
-         * @todo Make these arguments not nullable in v6
-         */
-        $this->db = $database;
-        $this->cache = $cache;
-        $this->cookie = $cookie;
     }
 
     /**
      * Static function to allow for method chaining without having to assign to a variable first.
      */
-    public static function new(IDatabase $database, ?ICache $cache = null, ?ICookie $cookie = null)
+    public static function new(IDatabase $db, ?ICache $cache = null, ?ICookie $cookie = null)
```

Comment:
`$cache` and `$cookie` are non-nullable as parameters in the `_construct()` function, but they're nullable here; should that be prevented?